src/matsim_io/writers.py

```diff
@@ -136,9 +136,10 @@ def add_activity_with_link(
             self._write(f' end_time="{self.time(end_time)}"')
         self._write("/>\n")
 
-    def add_leg_with_route(
+    def add_leg(
         self,
         route: list[Id],
+        mat_sim_routing: bool,
         mode: str = "car",
         departure_time: Optional[int] = None,
     ) -> None:
```

Comment:
Maybe add description of the mat_sim_routing parameter and change the description of the function. "Add a leg with a route to the plan." is not accurate anymore, there needs to be an if x...